internal/querycoord/cluster.go

```diff
@@ -457,46 +426,60 @@ func (c *queryNodeCluster) GetSegmentInfo(ctx context.Context, in *querypb.GetSe
 
 			segmentInfos = append(segmentInfos, segment)
 		}
-	} else {
-		allSegments := c.clusterMeta.showSegmentInfos(in.CollectionID, nil)
-		for _, segment := range allSegments {
-			if in.CollectionID == 0 || segment.CollectionID == in.CollectionID {
-				segmentInfos = append(segmentInfos, segment)
-			}
+		return segmentInfos, nil
+	}
+
+	allSegments := c.clusterMeta.showSegmentInfos(in.CollectionID, nil)
+	for _, segment := range allSegments {
+		if in.CollectionID == 0 || segment.CollectionID == in.CollectionID {
+			segmentInfos = append(segmentInfos, segment)
 		}
 	}
 
+	sealedSegmentNum := len(segmentInfos)
+
 	// Fetch growing segments
+	replicas, err := c.clusterMeta.getReplicasByCollectionID(in.CollectionID)
+	if err != nil {
+		return nil, err
+	}
+
 	c.RLock()
-	var wg sync.WaitGroup
-	cnt := len(c.nodes)
-	resChan := make(chan respTuple, cnt)
-	wg.Add(cnt)
-	for _, node := range c.nodes {
-		go func(node Node) {
-			defer wg.Done()
-			res, err := node.getSegmentInfo(ctx, in)
-			resChan <- respTuple{
-				res: res,
-				err: err,
+	var hasAvailableReplica bool
+	for _, replica := range replicas {
```

Comment:
Serial algorithm performance is slow？

Reply:
This loop stopped as it saw the first  available replica, and the number of replicas won't be large (<=3)